test/relayer_gaia_test.go

```diff
@@ -148,3 +158,164 @@ func TestGaiaReuseIdentifiers(t *testing.T) {
 	require.Equal(t, expectedSrc, src)
 	require.Equal(t, expectedDst, dst)
 }
+
+func TestGaiaMisbehaviourMonitoring(t *testing.T) {
+	chains := spinUpTestChains(t, gaiaChains...)
+
+	var (
+		src = chains.MustGet("ibc-0")
+		dst = chains.MustGet("ibc-1")
+	)
+
+	path, err := genTestPathAndSet(src, dst, "transfer", "transfer")
+	require.NoError(t, err)
+
+	// create path
+	_, err = src.CreateClients(dst)
+	require.NoError(t, err)
+	testClientPair(t, src, dst)
+
+	_, err = src.CreateOpenConnections(dst, 3, src.GetTimeout())
+	require.NoError(t, err)
+	testConnectionPair(t, src, dst)
+
+	_, err = src.CreateOpenChannels(dst, 3, src.GetTimeout())
+	require.NoError(t, err)
+	testChannelPair(t, src, dst)
+
+	// start the relayer process in it's own goroutine
+	rlyDone, err := relayer.RunStrategy(src, dst, path.MustGetStrategy())
+	require.NoError(t, err)
+
+	// Wait for relay message inclusion in both chains
+	require.NoError(t, src.WaitForNBlocks(1))
+	require.NoError(t, dst.WaitForNBlocks(1))
+
+	latestHeight, err := src.QueryLatestHeight()
+	require.NoError(t, err)
+
+	header, err := src.QueryHeaderAtHeight(latestHeight)
+	require.NoError(t, err)
+
+	clientStateRes, err := src.QueryClientState(latestHeight)
+	require.NoError(t, err)
+
+	// unpack any into ibc tendermint client state
+	clientStateExported, err := clienttypes.UnpackClientState(clientStateRes.ClientState)
+	require.NoError(t, err)
+
+	// cast from interface to concrete type
+	clientState, ok := clientStateExported.(*ibctmtypes.ClientState)
+	require.True(t, ok, "error when casting exported clientstate")
+
+	height := clientState.GetLatestHeight().(clienttypes.Height)
+	height.RevisionNumber = 0
```

Comment:
You shouldn't need to modify the `RevisionNumber`. That corresponds to the counterparty chain-id